### DISCUSSION TITLE
Account to feasibility and size constraints

### DISCUSSION THREAD

**Discussion by mandriuska** [OpShin/opshin#47] [Q&A] [answered]
I just started to look into eopsin, and it's great! Will definitely contribute in the short future.

Meanwhile, when updated from 0.7.8 --> 0.7.10 I found that 'Wrapper' smart contract size went from 6kb to 20kb. Do you know why it happened? It would help me to understand the depths better.

**Comment by nielstron** [comment]
Might simply be because a lot of the stdlib that is unused was not optimized out. Or because it uses str(int) which is a rather large function. Need to improve the dead variable remover/run it several times to see whats going on.

UPDATE: I tried running the dead variable optimizer several times, without any improvements. I tried removing str(int), without any improvement. Will further investigate.

**Comment by nielstron** [comment, marked as answer]
I found the issue. Its the resolution of this issue: https://github.com/ImperatorLang/eopsin/issues/43

Generally, assignments of names are not removed as they might throw a NameError. This leads to large parts of the prelude not being optimized away. Something like the following remains, even though neither `bytes` nor `Value` is ever accessed.

```python3
bytes = ... (injected builtin code)
TxId = bytes
```

**Reply by nielstron** [reply]
I will try to trim the optimizer to remove assignments that can not throw NameErrors (i.e. if they always occur in the same control flow after the name has been assigned)

**Reply by nielstron** [reply]
Version `0.7.11` should improve this (generated code went down to 15kb), but still 3x the previous size.

**Reply by nielstron** [reply]
A simple reason could also be that 0.7.8 was simply broken. The double function as redeemer/minter script was not correctly injected and the bytes constructor was missing. Still...

**Reply by nielstron** [reply]
[plutonomy-cli](https://github.com/ImperatorLang/plutonomy-cli) is still able to squench the script size by 60% to 6kb, so still room for optimizations...

**Comment by nielstron** [comment]
Update: I found an issue when introducing the double function as spend/mint script that would duplicate the main function, effectively doubling script size.

Has been resolved in Dev now
